src/einsteinpy/metric/kerr.py

```diff
@@ -1,72 +1,388 @@
+            (3, 3, 1),
+            (3, 3, 2),
+        ]
+
+        super().__init__(
+            coords=coords,
+            M=M,
+            a=a,
+            name="Kerr Metric",
+            metric_cov=self.metric_covariant,
+            christoffels=self._christoffels,
+            f_vec=self._f_vec,
         )
-        time_vel = kerr_utils.kerr_time_velocity(pos_vec, vel_vec, self.M, self.a.value)
 
-        self.initial_vec = np.hstack(
-            (self.time.value, pos_vec, time_vel.value, vel_vec)
+    def metric_covariant(self, x_vec):
+        """
+        Returns Covariant Kerr Metric Tensor \
+        in chosen Coordinates
+
+        Parameters
+        ----------
+        x_vec : ~numpy.ndarray
+            Position 4-Vector
+
+        Returns
+        -------
+        ~numpy.ndarray
+            Covariant Kerr Metric Tensor in chosen Coordinates
+            Numpy array of shape (4,4)
+
+        """
+        if self.coords == "KS":
+            return self._g_cov_ks(x_vec)
+
+        return self._g_cov_bl(x_vec)
+
+    def _g_cov_bl(self, x_vec):
+        """
+        Returns Covariant Kerr Metric Tensor \
+        in Boyer-Lindquist coordinates
+
+        Parameters
+        ----------
+        x_vec : ~numpy.ndarray
+            Position 4-Vector
+
+        Returns
+        -------
+        ~numpy.ndarray
+            Covariant Kerr Metric Tensor \
+            in Boyer-Lindquist coordinates
+            Numpy array of shape (4,4)
+
+        """
+        r, th = x_vec[1], x_vec[2]
+        r_s, M, a, c2 = self.sch_rad, self.M, self.a, _c ** 2
+        alpha = super().alpha(M, a)
+        sg, dl = super().sigma(r, th, M, a), super().delta(r, M, a)
+
+        g_cov_bl = np.zeros(shape=(4, 4), dtype=float)
+
+        g_cov_bl[0, 0] = 1 - (r_s * r / sg)
+        g_cov_bl[1, 1] = (sg / dl) * (-1 / c2)
+        g_cov_bl[2, 2] = -1 * sg / c2
+        g_cov_bl[3, 3] = (
+            (-1 / c2)
+            * (
+                (r ** 2)
+                + (alpha ** 2)
+                + (r_s * r * (np.sin(th) ** 2) * ((alpha ** 2) / sg))
+            )
+            * (np.sin(th) ** 2)
         )
-        self.scr = schwarzschild_radius(M)
+        g_cov_bl[0, 3] = g_cov_bl[3, 0] = (
+            r_s * r * alpha * (np.sin(th) ** 2) / (sg * _c)
+        )
+
+        return g_cov_bl
 
-    @classmethod
-    @u.quantity_input(time=u.s, M=u.kg, a=u.m)
-    def from_coords(cls, coords, M, q=None, Q=None, time=0 * u.s, a=0 * u.m):
+    def _g_cov_ks(self, x_vec):
         """
-        Constructor
+        Returns Covariant Kerr Metric Tensor \
+        in Kerr-Schild coordinates
+
+        Parameters
+        ----------
+        x_vec : ~numpy.ndarray
+            Position 4-Vector
+
+        Returns
+        -------
+        NotImplementedError
+            To be implemented after KS Coordinates
+
+        """
+        # To be implemented after KS Coordinates
+        raise NotImplementedError
+
+    def _dg_dx_bl(self, x_vec):
+        """
```

Comment:
Maybe use a more descriptive name for this function name

Reply:
Most of the code in these metric modules are from the now removed `<Metric>_utils` modules. So, I have tried to keep the function and variable names mostly the same, with changes only when the original name was confusing or a more descriptive name was available. In this case, this function was named `dmdx` earlier. In any case, these functions are not exposed to users and only used for calculations inside other functions. And, the docstring is descriptive. So, should I still go for a better name? This function is not supposed to be called by users, as denoted by `_` in front of the function name.